src/picasso-definition/interactions/pinch.js

```diff
@@ -25,11 +26,24 @@ const pinch = ({ chart, actions, viewHandler, rtl }) => ({
         return false;
       }
 
-      [this.pointArea] = chart
+      this.area = chart
         .componentsFromPoint({ x: e.center.x, y: e.center.y })
         .filter((c) => c.key === KEYS.COMPONENT.POINT || c.key === KEYS.COMPONENT.HEAT_MAP);
 
-      return this.pointArea;
+      if (!this.area.length) {
+        return false;
+      }
+
+      const rectSize = this.area[0]?.rect?.computedPhysical;
+
+      if (rectSize?.height && rectSize?.width) {
+        this.componentSize = { ...rectSize };
+        lastRectSize = { ...rectSize };
```

Comment:
if (rectSize?.height && rectSize?.width) {
lastRectSize = { ...rectSize };
}
this.componentSize = lastRectSize;

Reply:
Better if we do the following:
if (rectSize?.height && rectSize?.width) {
this.componentSize = { ...rectSize };
}